### PR TITLE
merge-tree: Add farm tests for zero length obliterate

### PR DESCRIPTION
**Pull request by jzaffiro**
Allow obliterate operations with a range of length zero in a separate set of tests to make debugging any issues easier. 

[AB#31182](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/31182)

### CODE REVIEW

File: packages/dds/merge-tree/src/test/obliterateOperations.ts [hunk 1]
```diff
@@ -114,20 +132,43 @@ export const obliterateField: TestOperation = (
...
+	random: IRandom,
+) => {
+	const fieldEndpoints = getFieldEndpoints(
+		client,
+		opStart,
+		// the operation runner generates endpoints with client length, but this model only supports up to client length - 1.
+		Math.min(opEnd, client.getLength() - 1),
+	);
+
+	let endISP: InteriorSequencePlace | undefined;
```
[the remark this answers is not in the document]
Comment (jzaffiro) [reply packages/dds/merge-tree/src/test/obliterateOperations.ts:167]: This variable is used on line 140.

**Review by Abe27342 — APPROVED** [review]
